processor/resourcedetectionprocessor/internal/aws/ec2/ec2.go

```diff
@@ -107,17 +108,19 @@ func getClientConfig(ctx context.Context, logger *zap.Logger) *http.Client {
 	return client
 }
 
-func connectAndFetchEc2Tags(region string, instanceID string, tagKeyRegexes []*regexp.Regexp, client *http.Client) (map[string]string, error) {
+func connectAndFetchEc2Tags(svc ec2iface.EC2API, region string, instanceID string, tagKeyRegexes []*regexp.Regexp, client *http.Client) (map[string]string, error) {
+	if svc != nil {
+		return fetchEC2Tags(svc, instanceID, tagKeyRegexes)
```

Comment:
This doesn't seem right.  The connect and fetch steps should probably be separated.  Having this function do two different things depending on whether one of its parameters is nil, and one of them being modifying that input parameter, seems wrong.  There should probably be a separate method to create the EC2 client that can then be used by the existing fetch function.

Reply:
Yeah I also had some qualms about this, is there a reason that there's a new session being created each time? I wanted to refactor this to create the client when we create the detector but figured there was a timeout or something to consider.

Reply:
and right even if keeping this structure the call to the second fetchEC2Tags probably should just use a new var like in the original code,
`` 
e = ec2.New(sess)
return fetchEC2Tags(e, instanceID, tagKeyRegexes)
``